include/velocypack/Slice.h

```diff
@@ -117,38 +117,70 @@ class Slice {
   // creates a slice of type MaxKey
   static constexpr Slice maxKeySlice() noexcept { return Slice(maxKeySliceData); }
 
-  // pointer to the head byte
-  constexpr uint8_t const* start() const noexcept { return _start; }
-  
+  int64_t getTagId() const {
+    if(isTagged()) {
+      if(*_start == 0xee) {
+        return readIntegerFixed<ValueLength, 1>(_start + 1);
+      } else if(*_start == 0xef) {
+        return readIntegerFixed<ValueLength, 8>(_start + 1);
+      } else {
+        throw new Exception(Exception::InternalError, "Invalid tag type ID");
+      }
+    }
+
+    return 0;
+  }
+
+  constexpr uint8_t tagOffset() const {
+    return tagOffset(rawStart());
+  }
+
+  // pointer to the head byte, including possible tags
+  constexpr uint8_t const* rawStart() const noexcept {
+    return _start;
+  }
+
+  // pointer to the head byte, excluding possible tags
+  constexpr uint8_t const* start() const noexcept {
+    return _start + tagOffset();
+  }
+
   // pointer to the head byte
   template <typename T>
   T const* startAs() const {
-    return reinterpret_cast<T const*>(_start);
+    return reinterpret_cast<T const*>(start());
```

Comment:
IMHO this should be reverted to just return (the converted pointer to) `_start` again, as this will ensure the old behavior that embedders may rely on.